### PR TITLE
Add hardknott (Yocto version) compatibility

### PR DESCRIPTION
**Pull request by jainvikas8**
# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ x] Make sure that all commits have a [`Changelog`](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags) tag. If nothing should be added to the Changelog, add a `Changelog: None` tag. If there is a change, add `Changelog: Commit`, or `Changelog: Title`, depending on what should be included in the changelog.

- [ x] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

The following changes are been done to support `hardknott`:
* Renamed `e2fsprogs` recipe to reflect minor version
* Convert layer overrides to new syntax format
* Update `layer.conf`

### CODE REVIEW

**Comment by jainvikas8** [comment]
@kacf I'm looking to raise Honistor related syntax changes in the next PR as they are dependent on this being approved.

File: meta-mender-core/recipes-bsp/grub/grub-efi-mender-precompiled_2.04.bb [hunk 1]
```diff
@@ -28,8 +28,6 @@ SRC_URI = " \
 
 S = "${WORKDIR}/git"
 
-require version_logic.inc
```
Comment (kacf) [inline meta-mender-core/recipes-bsp/grub/grub-efi-mender-precompiled_2.04.bb:31]: The commit message says this is unnecessary, but doesn't expand on why. This specifies several dependencies and conflicts for GRUB, so why remove it?

**Review by kacf — COMMENTED** [review]
Thanks, @dreedarm and @jainvikas8! The syntax change is quite a beast! I have left some comments below.

File: meta-mender-core/classes/mender-setup-image.inc [hunk 2]
```diff
@@ -8,7 +8,7 @@ IMAGE_FSTYPES += "${@bb.utils.contains('MENDER_FEATURES', 'mender-image-gpt', '
 MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_mender-image_x86 =     " kernel-image"
 MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_mender-image_x86-64 =  " kernel-image"
 MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_mender-image_arm =     " kernel-image kernel-devicetree"
-MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_mender-image_aarch64 = " kernel-image kernel-devicetree"
+MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_mender-image_aarch64 = " kernel-image"
```
[the remark this answers is not in the document]
Comment (dreedarm) [reply meta-mender-core/classes/mender-setup-image.inc]: Not always,  When using EDK2 it's more normal to use ACPI tables and that's how our platform currently works.  At some point we may change it to use EDK2 but with a devicetree config.

File: meta-mender-core/recipes-bsp/grub/grub-efi-mender-precompiled_2.04.bb (hunk 1 above)
Comment (dreedarm) [reply meta-mender-core/recipes-bsp/grub/grub-efi-mender-precompiled_2.04.bb:31]: Just checked the code and the reason we removed it is that the whole file assumes grub is booted from u-boot and it's u-boot version conflicts with grub that is being searched for but we don't use u-boot so none of it helps.

Not really sure how to change it to work with both u-boot and edk2, any suggestions?

File: meta-mender-core/classes/mender-part-images.bbclass [hunk 3]
```diff
@@ -270,7 +270,6 @@ _MENDER_PART_IMAGE_DEPENDS += "${@bb.utils.contains('MENDER_DATA_PART_FSTYPE', '
 # This assumes that U-boot is used on ARM, this could become problematic
 # if we add support for other bootloaders on ARM, e.g Barebox.
 _MENDER_PART_IMAGE_DEPENDS_append_mender-grub_arm =     " u-boot:do_deploy"
-_MENDER_PART_IMAGE_DEPENDS_append_mender-grub_aarch64 = " u-boot:do_deploy"
```
[the remark this answers is not in the document]
Comment (dreedarm) [reply meta-mender-core/classes/mender-part-images.bbclass:273]: Not sure what to do here to tell what bootloader is in use or if using devicetree / ACPI configs.  Have asked the meta-arm maintainers if they know a way to check these questions.

[the remark this answers is not in the document]
Comment (jainvikas8) [reply meta-mender-core/classes/mender-part-images.bbclass:273]: Update: After discussing with the maintainers, the suggestion was to use a virtual bootloader to figure out whether u-boot or EDK2 was used and check `KERNEL_DEVICETREE` to see if devicetree is required.

[the remark this answers is not in the document]
Comment (kacf) [reply meta-mender-core/classes/mender-part-images.bbclass:273]: You mean like [`virtual/bootloader`](https://docs.yoctoproject.org/singleindex.html#using-virtual-providers)? I think this is set to "grub-efi" when GRUB is used as UEFI app. It doesn't really describe which bootloader is used as the UEFI provider.

I'm fine with leaving it as you suggested for now. It sounds like there is no consensus for how to handle the two-stage bootloader scenario in Yocto yet, where you have "UEFI provider -> UEFI app".

File: meta-mender-core/classes/mender-setup-image.inc (hunk 2 above)
[the remark this answers is not in the document]
Comment (kacf) [reply meta-mender-core/classes/mender-setup-image.inc]: Ok, got it.

**Comment by jainvikas8** [comment]
> @jainvikas8: In the end the discussions above resulted in no changes. Anything else you want to do here, or shall I go ahead and merge?

Are we good with the `hardknott` patch, can we run some pipeline?

**Comment by jainvikas8** [comment]
Force pushed to add more information to commit message for u-boot patch

**Comment by kacf** [comment]
> Are we good with the `hardknott` patch, can we run some pipeline?

I'd like to, but we don't have the infrastructure to run the post-dunfell branches yet. It requires changes in other repositories too. I'll trust our eyes for now, and I will pick up the work on the pipelines once Kirkstone drops in April. It will be the next officially supported branch by Mender.

**Pull request merged**